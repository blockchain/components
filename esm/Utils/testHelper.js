import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}