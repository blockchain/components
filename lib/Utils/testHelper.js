"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}