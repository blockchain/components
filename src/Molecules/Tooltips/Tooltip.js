import React from 'react'
import PropTypes from 'prop-types'
import styled, { injectGlobal } from 'styled-components'
import ReactTooltip from 'react-tooltip'

injectGlobal`
  .__react_component_tooltip {
    border-radius: 3px;
    display: inline-block;
    font-size: 13px;
    left: -999em;
    opacity: 0;
    padding: 8px 21px;
    position: fixed;
    pointer-events: none;
    transition: opacity 0.3s ease-out;
    top: -999em;
    visibility: hidden;
    z-index: 999;

    &::before,
    &::after {
      content: "";
      width: 0;
      height: 0;
      position: absolute;
    }

    &.show {
      opacity: 0.9;
      margin-top: 0;
      margin-left: 0;
      visibility: visible;
    }

    &.place-top {
      margin-top: -10px;

      &::before {
        border-left: 10px solid transparent;
        border-right: 10px solid transparent;
        bottom: -8px;
        left: 50%;
        margin-left: -10px;
      }

      &::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
      }
    }

    &.place-bottom {
      margin-top: 10px;

      &::before {
        border-left: 10px solid transparent;
        border-right: 10px solid transparent;
        top: -8px;
        left: 50%;
        margin-left: -10px;
      }

      &::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
      }
    }

    &.place-left {
      margin-left: -10px;

      &::before {
        border-top: 6px solid transparent;
        border-bottom: 6px solid transparent;
        right: -8px;
        top: 50%;
        margin-top: -5px;
      }

      &::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
      }
    }

    &.place-right {
      margin-left: 10px;

      &::before {
        border-top: 6px solid transparent;
        border-bottom: 6px solid transparent;
        left: -8px;
        top: 50%;
        margin-top: -5px;
      }

      &::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
      }
    }

    .multi-line {
      display: block;
      padding: 2px 0;
      text-align: center;
    }
  }
`

const StyledTip = styled(ReactTooltip)`
  color: ${(props) => props.theme['tooltipColor']} !important;
  border: 1px solid ${(props) => props.theme['tooltipBorderColor']} !important;
  background-color: ${(props) =>
    props.theme['tooltipBackgroundColor']} !important;
  cursor: pointer;
  max-width: 350px;
  font-family: ${(props) => props.theme['fontPrimary']};
  font-size: 0.8rem;
  font-weight: 300;
  z-index: 5;
  text-transform: none;
  text-align: left;
  pointer-events: auto !important;

  &:hover {
    visibility: visible !important;
    opacity: 1 !important;
  }

  &.place-top {
    &::before {
      border-top-color: ${(props) =>
        props.theme['tooltipBorderColor']} !important;
    }

    &::after {
      border-top-color: ${(props) =>
        props.theme['tooltipBackgroundColor']} !important;
    }
  }

  &.place-left {
    &::before {
      border-left-color: ${(props) =>
        props.theme['tooltipBorderColor']} !important;
    }

    &::after {
      border-left-color: ${(props) =>
        props.theme['tooltipBackgroundColor']} !important;
    }
  }

  &.place-bottom {
    &::before {
      border-bottom-color: ${(props) =>
        props.theme['tooltipBorderColor']} !important;
    }

    &::after {
      border-bottom-color: ${(props) =>
        props.theme['tooltipBackgroundColor']} !important;
    }
  }

  &.place-right {
    &::before {
      border-right-color: ${(props) =>
        props.theme['tooltipBorderColor']} !important;
    }

    &::after {
      border-right-color: ${(props) =>
        props.theme['tooltipBackgroundColor']} !important;
    }
  }
`

const Tooltip = (props) => {
  const { id, children, ...rest } = props

  return (
    <StyledTip
      border
      delayHide={250}
      effect="solid"
      id={id}
      type="light"
      {...rest}
    >
      {children}
    </StyledTip>
  )
}

Tooltip.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
}

export default Tooltip
