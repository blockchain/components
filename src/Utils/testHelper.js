import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
