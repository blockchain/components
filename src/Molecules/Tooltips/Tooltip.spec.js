import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Tooltip from './Tooltip'

describe('Tooltip', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Tooltip id="storybook.test">Tooltip</Tooltip>)

    expect(tree).toMatchSnapshot()
  })
})
