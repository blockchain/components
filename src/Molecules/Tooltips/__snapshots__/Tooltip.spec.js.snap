// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render correctly 1`] = `
.c0 {
  color: #FFF !important;
  border: 1px solid #383838 !important;
  background-color: #383838 !important;
  cursor: pointer;
  max-width: 350px;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 0.8rem;
  font-weight: 300;
  z-index: 5;
  text-transform: none;
  text-align: left;
  pointer-events: auto !important;
}

.c0:hover {
  visibility: visible !important;
  opacity: 1 !important;
}

.c0.place-top::before {
  border-top-color: #383838 !important;
}

.c0.place-top::after {
  border-top-color: #383838 !important;
}

.c0.place-left::before {
  border-left-color: #383838 !important;
}

.c0.place-left::after {
  border-left-color: #383838 !important;
}

.c0.place-bottom::before {
  border-bottom-color: #383838 !important;
}

.c0.place-bottom::after {
  border-bottom-color: #383838 !important;
}

.c0.place-right::before {
  border-right-color: #383838 !important;
}

.c0.place-right::after {
  border-right-color: #383838 !important;
}

<Tooltip
  id="storybook.test"
>
  <Styled(ReactTooltip)
    border={true}
    delayHide={250}
    effect="solid"
    id="storybook.test"
    type="light"
  >
    <ReactTooltip
      border={true}
      className="c0"
      delayHide={250}
      effect="solid"
      id="storybook.test"
      insecure={true}
      resizeHide={true}
      type="light"
      wrapper="div"
    >
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
        id="storybook.test"
      >
        Tooltip
      </div>
    </ReactTooltip>
  </Styled(ReactTooltip)>
</Tooltip>
`;
